### PR TITLE
Updates `Relaxation` to work with grids with `Flat` dimensions

### PR DESCRIPTION
**Pull request by jagoosw**
Added some methods to allow relaxation forcing to work again.

### CODE REVIEW

File: src/Forcings/relaxation.jl [hunk 1]
```diff
@@ -83,6 +83,19 @@ end
...
+    f.rate * f.mask(x₁, x₂) * (f.target(x₁, x₂, t) - field)
+
+@inline (f::Relaxation{R, M, <:Number})(x₁, x₂, t, field) where {R, M} =
+    f.rate * f.mask(x₁, x₂) * (f.target - field)
+
+@inline (f::Relaxation)(x₁, t, field) =
+    f.rate * f.mask(x₁) * (f.target(x₁, t) - field)
+
+@inline (f::Relaxation{R, M, <:Number})(x₁, t, field) where {R, M} =
+    f.rate * f.mask(x₁) * (f.target - field)
```
[the remark this answers is not in the document]
Comment (jagoosw) [reply src/Forcings/relaxation.jl:97]: Ah yes I'll change to that notation, I couldn't remember what the standard was

[the remark this answers is not in the document]
Comment (glwagner) [reply src/Forcings/relaxation.jl:97]: The convention is introduced here: https://github.com/CliMA/Oceananigans.jl/blob/615b2a854e86cc57f3e900eb463a7fe1fe890f1d/src/Grids/nodes_and_spacings.jl#L31

[the remark this answers is not in the document]
Comment (glwagner) [reply src/Forcings/relaxation.jl:97]: Hmm, I see what you're saying. We don't need to define functions for both `ξ, t` and `η, t`, for example. Maybe it is best to keep x1, x2 after all. A comment might help too to clarify that `x1, t` can represent either `ξ, t` or `η, t`.

[the remark this answers is not in the document]
Comment (jagoosw) [reply src/Forcings/relaxation.jl:100]: Is what I've added okay?

File: src/Forcings/relaxation.jl [hunk 2]
```diff
@@ -83,6 +83,22 @@ end
 @inline (f::Relaxation{R, M, <:Number})(x, y, z, t, field) where {R, M} =
     f.rate * f.mask(x, y, z) * (f.target - field)
 
+# methods for grids with flat dimensions
+# here xₙ is used as the corrdinate they represent depends on which dimension is flat
+# for example in the below method (x₁, x₂) may be (ξ, η), (ξ, r), or (η, r)
```
Comment (glwagner) [inline src/Forcings/relaxation.jl:88]:
```suggestion
# Methods for grids with Flat dimensions:
# Here, the meaning of the coordinate xₙ depends on which dimension is Flat:
# for example, in the below method (x₁, x₂) may be (ξ, η), (ξ, r), or (η, r).
```

File: src/Forcings/relaxation.jl [hunk 3]
```diff
@@ -83,6 +83,22 @@ end
...
+@inline (f::Relaxation)(x₁, x₂, t, field) =
+    f.rate * f.mask(x₁, x₂) * (f.target(x₁, x₂, t) - field)
+
+@inline (f::Relaxation{R, M, <:Number})(x₁, x₂, t, field) where {R, M} =
+    f.rate * f.mask(x₁, x₂) * (f.target - field)
+
+# and here x₁ can be ξ, η, or r
```
Comment (glwagner) [inline src/Forcings/relaxation.jl:95]:
```suggestion
# Below, the coordinate x₁ can be ξ, η, or r (see above)
```

**Review by glwagner — APPROVED** [review]
Great! Just made some suggestions to punctuate properly and add a bit more explanation.

File: src/Forcings/relaxation.jl (hunk 1 above)
[the remark this answers is not in the document]
Comment (glwagner) [reply src/Forcings/relaxation.jl:100]: It looks good! Only suggestion is to write comments the same way we typically write when we want to be careful and clear; ie use punctuation and captalization.

**Comment by jagoosw** [comment]
Hmm I just hit merge but then thought I should probably have bumped the patch number right?

**Pull request merged**